### ISSUE TITLE
Not setting Watched status on first playthrough.

### ISSUE THREAD

**Issue by Laurvin** [maximeh/script.myepisodes#23]
I'm afraid I'm back with an issue...

I've opened a new one cause I am not sure if it is related to the other one.

What's happening is that when I watch any show, the script runs and shows the title of the show if it recognizes it. But at the end, whether I use the stop button, let it run its course, or use seek to go to the end, it will just not update the Watched status. If I then open the same file again, use seek to go a number of minutes forward and then end it by any of the three ways above, it *will* correctly update the Watched status.

This has been happening with all shows I have been watching with the new version.

I've uploaded a Kodi log where I watch a show normally and then go through it fast: https://raw.githubusercontent.com/Laurvin/Temp/main/Log

Hope you can make sense of this.

Thanks!

**Comment by maximeh** [comment]
Oh that's very interesting, looks like the connection to the server gets closed when it's time to mark it as watched. I'll have a look at that.

**Comment by maximeh** [comment]
Heya; sorry for the delay there, could you try with that one? That should fix this issue.
TL;DR: The server closes the connection because it's open too long and there was no retry mechanism in place.
[script.myepisodes-3.1.1.zip](https://github.com/maximeh/script.myepisodes/files/11186244/script.myepisodes-3.1.1.zip)

**Comment by Laurvin** [comment]
Thanks, will try tomorrow!

**Comment by Laurvin** [comment]
I've tested two episodes and so far it's working flawlessly!

**Comment by maximeh** [comment]
Well I'm glad, I'll put out the code change soon and make a new release then. Hopefully it'll work this time for you for some time :)

**Comment by maximeh** [comment]
I'll close this and I will submit a proper update to Kodi.

Thanks for your testing and reporting, much appreciated.

**Comment by Laurvin** [comment]
Glad to help. Thanks for writing this add-on!
